interoperator/pkg/webhook/default_server/sfserviceinstance/mutating/sfserviceinstance_create_handler_test.go

```diff
@@ -0,0 +1,214 @@
+				t.Errorf("SFServiceInstanceCreateHandler.mutatingSFServiceInstanceFn() error = %v, wantErr %v", err, tt.wantErr)
+			}
+			// TODO: check for modified labels
+		})
+	}
+}
+
+func TestSFServiceInstanceCreateHandler_Handle(t *testing.T) {
+	type fields struct {
+		Decoder types.Decoder
+	}
+	type args struct {
+		ctx context.Context
+		req types.Request
+	}
+	decoder, err := admission.NewDecoder(runtime.NewScheme())
+	if err != nil {
+		t.Error(err)
+	}
+	parameters := `{
+		"foo": "bar",
+		"baz": {
+			"fizz": "buzz"
+		}
+	}`
+	re := &runtime.RawExtension{}
+	_ = re.UnmarshalJSON([]byte(parameters))
+	spec := osbv1alpha1.SFServiceInstanceSpec{
+		ServiceID:        "service-id",
+		PlanID:           "plan-id",
+		RawContext:       re,
+		OrganizationGUID: "org-id",
+		SpaceGUID:        "space-id",
+		RawParameters:    re,
+		PreviousValues:   re,
+	}
+	created := &osbv1alpha1.SFServiceInstance{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "foo",
+			Namespace: "default",
+		},
+		Spec: spec,
+		Status: osbv1alpha1.SFServiceInstanceStatus{
+			DashboardURL: "",
+			State:        "",
+			Error:        "",
+			Description:  "",
+			AppliedSpec:  spec,
+		},
+	}
+	req := types.Request{
+		AdmissionRequest: &admissionv1beta1.AdmissionRequest{
+			UID:         "",
+			Kind:        metav1.GroupVersionKind{},
+			Resource:    metav1.GroupVersionResource{},
+			SubResource: "",
+			Name:        "",
+			Namespace:   "",
+			Operation:   "",
+			UserInfo:    authenticationv1.UserInfo{},
+			Object: runtime.RawExtension{
+				Raw: func() []byte {
+					buf := new(bytes.Buffer)
+					json.NewEncoder(buf).Encode(created)
+					return buf.Bytes()
+				}(),
+				Object: created,
+			},
+			OldObject: runtime.RawExtension{},
+			DryRun: func() *bool {
+				dryrun := false
+				return &dryrun
+			}(),
+		}}
+	resp := types.Response{
+		Patches: []jsonpatch.JsonPatchOperation{{
+			Operation: "add",
+			Path:      "/metadata/labels",
+			// Since the value corresponding to "baz" in `spec.Context` is not
+			// a string, it will not be added as a label and hence will be
+			// absent in the resulting admission response.
+			Value:     map[string]string{"foo": "bar"},
+		}},
+		Response: &admissionv1beta1.AdmissionResponse{
+			UID:     "",
+			Allowed: true,
+			Result:  nil,
+			Patch:   nil,
+			PatchType: func() *admissionv1beta1.PatchType {
+				pt := admissionv1beta1.PatchTypeJSONPatch
+				return &pt
+			}(),
+			AuditAnnotations: map[string]string{},
+		},
+	}
+	tests := []struct {
+		name   string
+		fields fields
+		args   args
+		want   types.Response
+	}{
+		{
+			"Test to validate admission request handling",
+			fields{decoder},
+			args{context.TODO(), req},
+			resp,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			h := &SFServiceInstanceCreateHandler{
+				Decoder: tt.fields.Decoder,
+			}
+			if got := h.Handle(tt.args.ctx, tt.args.req); !reflect.DeepEqual(got, tt.want) {
```

Comment:
The underlying issue was due to difference in types deep inside `types.Response.jsonpatch.JsonPatchOperation` -
```diff
  types.Response{
  	Patches: []jsonpatch.Operation{
  		{
  			Operation: "add",
  			Path:      "/metadata/labels",
- 			Value:     map[string]interface{}{"foo": string("bar")},
+ 			Value:     map[string]string{"foo": "bar"},
  		},
  	},
  	Response: &v1beta1.AdmissionResponse{
  		... // 3 identical fields
  		Patch:            nil,
  		PatchType:        &"JSONPatch",
- 		AuditAnnotations: nil,
+ 		AuditAnnotations: map[string]string{},
  	},
  }
```
In retrospect, I should've checked the type information recursively instead of just checking the first level entities.
@vivekzhere  @jayeshjk